### PR TITLE
KOGITO-1943 - Add integration tests with Infinispan

### PR DESCRIPTION
**Pull request by cristianonicolai**

### CODE REVIEW

**Comment by jstastny-cz** [comment]
I thought we're sticking to testcontainers for running the docker containers. The docker-maven-plugin seems to have the same limitation[1] that testcontainers do [2] - supports docker only.

One of the arguments against mavenized docker container startup was that when using testcontainers it's plain Java - the tests can be run from IDE out of the box.

[1] https://github.com/fabric8io/docker-maven-plugin/issues/1330
[2] https://github.com/testcontainers/testcontainers-java/issues/2088

**Comment by cristianonicolai** [comment]
@jstastny-cz reviewed tests

File: process-infinispan-persistence-springboot/src/test/java/org/acme/deals/DealsRestIT.java [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+    @Container
+    public static GenericContainer INFINISPAN = new FixedHostPortGenericContainer(System.getProperty("container.image.infinispan"))
+            .withFixedExposedPort(11222, 11222)
+            .withEnv("USER", "admin")
+            .withEnv("PASS", "admin")
+            .waitingFor(Wait.forLogMessage(".*ISPN080001.*", 1));
+
+    @BeforeAll
+    public static void setup() {
+        RestAssured.port = 8080;
```
Comment (jstastny-cz) [inline process-infinispan-persistence-springboot/src/test/java/org/acme/deals/DealsRestIT.java]: I believe the port can be injected as a field using annotation @LocalServerPort, that removes the dependency of the test on configuration. Similarly to process-usertasks-with-security-oidc-springboot

```
   @LocalServerPort
    int randomServerPort;

    @BeforeAll
    public void before() {
        RestAssured.port = randomServerPort;
    }
```

**Review by jstastny-cz — APPROVED** [review]
Looks good, just a minor comment regarding spring boot port handling.

File: process-infinispan-persistence-springboot/src/test/java/org/acme/deals/DealsRestIT.java (hunk 1 above)
Comment (cristianonicolai) [reply process-infinispan-persistence-springboot/src/test/java/org/acme/deals/DealsRestIT.java]: @jstastny-cz done

**Pull request merged**